external/llvm-project/mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -174,7 +174,7 @@ struct RawBufferOpLowering : public ConvertOpToLLVMPattern<GpuOp> {
     uint32_t word3 = (7 << 12) | (4 << 15);
     if (chipset.majorVersion == 10) {
       word3 |= (1 << 24);
-      uint32_t oob = adaptor.boundsCheck() ? 1 : 2;
+      uint32_t oob = adaptor.boundsCheck() ? 3 : 2;
       word3 |= (oob << 28);
```

Comment:
Just trying to understand, I'm confused on: In above comments you mentioned this out of bounds select is RDNA only, now instead of base this bit on architecture, this is based on whether or not we perform bounds check. Why is that? And how does it impact non-RDNA arch?

Reply:
RDNA has an `OOB_SELECT` field - 2 = no checks, 3 = `offset + lengthOfOp > numRecords - sgproffset` . CDNA has its oob checks inferred from the parameters to the instruction and no way to outright disable them

Reply:
If that's the case, could you amend your comments in 171 to incorporate those? Right now the comment still points to `3 = swizzles` which is much less clear compared to your previous comment.